WebGL has a lot of #if USE(ANGLE) blocks, making it hard to maintain

kkinnunen-apple opened the pull request:
#### d257ea20c5996ada3c009a6a2f4a639f92c0e2ca
<pre>
WebGL has a lot of #if USE(ANGLE) blocks, making it hard to maintain
<a href="https://bugs.webkit.org/show_bug.cgi?id=229405">https://bugs.webkit.org/show_bug.cgi?id=229405</a>
rdar://problem/82517622

Reviewed by Žan Doberšek.

Remove non-ANGLE WebGL implementations. This makes ANGLE the only WebGL backend
available. WebGL specific alterations to OpenGL semantics is implemented
correctly only in ANGLE. The OpenGL and OpenGL ES backends were not
being updated to match the WebGL specification or passing the tests.

Remove USE_ANGLE from WebKit level. Use USE_ANGLE for the ports
that want to control whether OpenGL is ANGLE or OpenGL is the platform
OpenGL.

ENABLE(WEBGL) &amp;&amp; USE(ANGLE) is now ENABLE(WEBGL).

* Source/ThirdParty/ANGLE/CMakeLists.txt:
* Source/ThirdParty/ANGLE/PlatformGTK.cmake:
* Source/ThirdParty/ANGLE/PlatformWPE.cmake:
* Source/ThirdParty/ANGLE/include/CMakeLists.txt:
* Source/WTF/wtf/Platform.h:
* Source/WebCore/CMakeLists.txt:
* Source/WebCore/Modules/webxr/WebXROpaqueFramebuffer.cpp:
(WebCore::WebXROpaqueFramebuffer::startFrame):
(WebCore::WebXROpaqueFramebuffer::endFrame):
(WebCore::WebXROpaqueFramebuffer::setupFramebuffer):
* Source/WebCore/Modules/webxr/WebXRWebGLLayer.cpp:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/html/canvas/ANGLEInstancedArrays.cpp:
(WebCore::ANGLEInstancedArrays::supported):
* Source/WebCore/html/canvas/WebGLCompressedTextureS3TC.cpp:
(WebCore::WebGLCompressedTextureS3TC::supported):
* Source/WebCore/html/canvas/WebGLDrawBuffers.cpp:
(WebCore::WebGLDrawBuffers::supported):
* Source/WebCore/html/canvas/WebGLFramebuffer.cpp:
(WebCore::WebGLFramebuffer::setAttachmentForBoundFramebuffer):
(WebCore::WebGLFramebuffer::removeAttachmentFromBoundFramebuffer):
(WebCore::getClearBitsByAttachmentType): Deleted.
(WebCore::getClearBitsByFormat): Deleted.
(WebCore::isAttachmentComplete): Deleted.
(WebCore::WebGLFramebuffer::getColorBufferWidth const): Deleted.
(WebCore::WebGLFramebuffer::getColorBufferHeight const): Deleted.
(WebCore::WebGLFramebuffer::getColorBufferFormat const): Deleted.
(WebCore::WebGLFramebuffer::checkStatus const): Deleted.
(WebCore::WebGLFramebuffer::onAccess): Deleted.
(WebCore::WebGLFramebuffer::initializeAttachments): Deleted.
* Source/WebCore/html/canvas/WebGLFramebuffer.h:
* Source/WebCore/html/canvas/WebGLProgram.cpp:
(WebCore::WebGLProgram::cacheActiveAttribLocations):
* Source/WebCore/html/canvas/WebGLRenderingContext.cpp:
(WebCore::WebGLRenderingContext::initializeVertexArrayObjects):
* Source/WebCore/html/canvas/WebGLRenderingContextBase.cpp:
(WebCore::WebGLRenderingContextBase::initializeNewContext):
(WebCore::WebGLRenderingContextBase::setupFlags):
(WebCore::WebGLRenderingContextBase::~WebGLRenderingContextBase):
(WebCore::WebGLRenderingContextBase::reshape):
(WebCore::WebGLRenderingContextBase::bindTexture):
(WebCore::WebGLRenderingContextBase::checkFramebufferStatus):
(WebCore::WebGLRenderingContextBase::clear):
(WebCore::WebGLRenderingContextBase::compileShader):
(WebCore::WebGLRenderingContextBase::compressedTexImage2D):
(WebCore::WebGLRenderingContextBase::compressedTexSubImage2D):
(WebCore::WebGLRenderingContextBase::copyTexSubImage2D):
(WebCore::WebGLRenderingContextBase::deleteTexture):
(WebCore::WebGLRenderingContextBase::disable):
(WebCore::WebGLRenderingContextBase::disableVertexAttribArray):
(WebCore::WebGLRenderingContextBase::validateVertexAttributes):
(WebCore::WebGLRenderingContextBase::drawArrays):
(WebCore::WebGLRenderingContextBase::drawElements):
(WebCore::WebGLRenderingContextBase::enable):
(WebCore::WebGLRenderingContextBase::enableVertexAttribArray):
(WebCore::WebGLRenderingContextBase::finish):
(WebCore::WebGLRenderingContextBase::flush):
(WebCore::WebGLRenderingContextBase::framebufferRenderbuffer):
(WebCore::WebGLRenderingContextBase::framebufferTexture2D):
(WebCore::WebGLRenderingContextBase::generateMipmap):
(WebCore::WebGLRenderingContextBase::getProgramParameter):
(WebCore::WebGLRenderingContextBase::getUniform):
(WebCore::WebGLRenderingContextBase::getUniformLocation):
(WebCore::WebGLRenderingContextBase::getVertexAttrib):
(WebCore::WebGLRenderingContextBase::linkProgramWithoutInvalidatingAttribLocations):
(WebCore::WebGLRenderingContextBase::readPixels):
(WebCore::WebGLRenderingContextBase::renderbufferStorage):
(WebCore::WebGLRenderingContextBase::shaderSource):
(WebCore::WebGLRenderingContextBase::texImageSourceHelper):
(WebCore::WebGLRenderingContextBase::texImageArrayBufferViewHelper):
(WebCore::WebGLRenderingContextBase::texImageImpl):
(WebCore::WebGLRenderingContextBase::texImage2DBase):
(WebCore::WebGLRenderingContextBase::texSubImage2DBase):
(WebCore::WebGLRenderingContextBase::validateTexFunc):
(WebCore::WebGLRenderingContextBase::validateTexFuncParameters):
(WebCore::WebGLRenderingContextBase::validateTexFuncFormatAndType):
(WebCore::WebGLRenderingContextBase::validateForbiddenInternalFormats):
(WebCore::WebGLRenderingContextBase::copyTexImage2D):
(WebCore::WebGLRenderingContextBase::texParameter):
(WebCore::WebGLRenderingContextBase::uniform1i):
(WebCore::WebGLRenderingContextBase::uniform1iv):
(WebCore::WebGLRenderingContextBase::validateTextureBinding):
(WebCore::WebGLRenderingContextBase::validateTexture2DBinding):
(WebCore::WebGLRenderingContextBase::vertexAttribfImpl):
(WebCore::WebGLRenderingContextBase::vertexAttribfvImpl):
(WebCore::WebGLRenderingContextBase::setFramebuffer):
(WebCore::WebGLRenderingContextBase::restoreCurrentTexture2D):
(WebCore::WebGLRenderingContextBase::drawArraysInstanced):
(WebCore::WebGLRenderingContextBase::drawElementsInstanced):
(WebCore::clip2D): Deleted.
(WebCore::getChannelBitsByFormat): Deleted.
(WebCore::possibleFormatAndTypeForInternalFormat): Deleted.
(WebCore::WebGLRenderingContextBase::validateNPOTTextureLevel): Deleted.
(WebCore::WebGLRenderingContextBase::validateDrawArrays): Deleted.
(WebCore::WebGLRenderingContextBase::validateDrawElements): Deleted.
(WebCore::internalFormatTheme): Deleted.
(WebCore::numberOfComponentsForFormat): Deleted.
(WebCore::numberOfComponentsForInternalFormat): Deleted.
(WebCore::isRGBFormat): Deleted.
(WebCore::WebGLRenderingContextBase::checkTextureCompleteness): Deleted.
(WebCore::WebGLRenderingContextBase::createFallbackBlackTextures1x1): Deleted.
(WebCore::WebGLRenderingContextBase::isTexInternalFormatColorBufferCombinationValid): Deleted.
(WebCore::WebGLRenderingContextBase::getBoundReadFramebufferColorFormat): Deleted.
(WebCore::WebGLRenderingContextBase::getBoundReadFramebufferWidth): Deleted.
(WebCore::WebGLRenderingContextBase::getBoundReadFramebufferHeight): Deleted.
(WebCore::calculateBytesForASTC): Deleted.
(WebCore::WebGLRenderingContextBase::validateCompressedTexFuncData): Deleted.
(WebCore::WebGLRenderingContextBase::validateCompressedTexDimensions): Deleted.
(WebCore::WebGLRenderingContextBase::validateCompressedTexSubDimensions): Deleted.
(WebCore::WebGLRenderingContextBase::initVertexAttrib0): Deleted.
(WebCore::WebGLRenderingContextBase::validateSimulatedVertexAttrib0): Deleted.
(WebCore::WebGLRenderingContextBase::simulateVertexAttrib0): Deleted.
(WebCore::WebGLRenderingContextBase::restoreStatesAfterVertexAttrib0Simulation): Deleted.
(WebCore::WebGLRenderingContextBase::applyStencilTest): Deleted.
* Source/WebCore/html/canvas/WebGLRenderingContextBase.h:
* Source/WebCore/html/canvas/WebGLShader.h:
* Source/WebCore/html/canvas/WebGLTexture.cpp:
(WebCore::WebGLTexture::WebGLTexture):
(WebCore::WebGLTexture::setTarget):
(WebCore::WebGLTexture::setParameteri): Deleted.
(WebCore::WebGLTexture::setParameterf): Deleted.
(WebCore::WebGLTexture::setLevelInfo): Deleted.
(WebCore::WebGLTexture::generateMipmapLevelInfo): Deleted.
(WebCore::WebGLTexture::getInternalFormat const): Deleted.
(WebCore::WebGLTexture::getType const): Deleted.
(WebCore::WebGLTexture::getWidth const): Deleted.
(WebCore::WebGLTexture::getHeight const): Deleted.
(WebCore::WebGLTexture::isValid const): Deleted.
(WebCore::WebGLTexture::markInvalid): Deleted.
(WebCore::WebGLTexture::isNPOT): Deleted.
(WebCore::WebGLTexture::isNPOT const): Deleted.
(WebCore::WebGLTexture::needToUseBlackTexture const): Deleted.
(WebCore::WebGLTexture::canGenerateMipmaps): Deleted.
(WebCore::WebGLTexture::isCompressed const): Deleted.
(WebCore::WebGLTexture::setCompressed): Deleted.
(WebCore::WebGLTexture::mapTargetToIndex const): Deleted.
(WebCore::internalFormatIsFloatType): Deleted.
(WebCore::internalFormatIsHalfFloatType): Deleted.
(WebCore::WebGLTexture::update): Deleted.
(WebCore::WebGLTexture::getLevelInfo const): Deleted.
* Source/WebCore/html/canvas/WebGLTexture.h:
* Source/WebCore/html/canvas/WebGLVertexArrayObjectBase.cpp:
(WebCore::WebGLVertexArrayObjectBase::WebGLVertexArrayObjectBase):
(WebCore::WebGLVertexArrayObjectBase::unbindBuffer):
(WebCore::WebGLVertexArrayObjectBase::updateVertexAttrib0): Deleted.
* Source/WebCore/html/canvas/WebGLVertexArrayObjectBase.h:
* Source/WebCore/platform/TextureMapper.cmake:
* Source/WebCore/platform/graphics/ANGLEWebKitBridge.cpp: Removed.
* Source/WebCore/platform/graphics/ANGLEWebKitBridge.h: Removed.
* Source/WebCore/platform/graphics/GraphicsContextGL.cpp:
* Source/WebCore/platform/graphics/angle/ANGLEHeaders.h:
* Source/WebCore/platform/graphics/angle/ANGLEUtilities.cpp:
* Source/WebCore/platform/graphics/angle/ANGLEUtilities.h:
* Source/WebCore/platform/graphics/angle/GraphicsContextGLANGLE.cpp:
* Source/WebCore/platform/graphics/angle/GraphicsContextGLANGLE.h:
* Source/WebCore/platform/graphics/cg/GraphicsContextGLCG.cpp:
* Source/WebCore/platform/graphics/cocoa/ANGLEUtilitiesCocoa.cpp:
* Source/WebCore/platform/graphics/cocoa/ANGLEUtilitiesCocoa.h:
* Source/WebCore/platform/graphics/gbm/GraphicsContextGLANGLELinux.cpp:
* Source/WebCore/platform/graphics/gbm/GraphicsContextGLFallback.cpp:
* Source/WebCore/platform/graphics/gbm/GraphicsContextGLFallback.h:
* Source/WebCore/platform/graphics/gbm/GraphicsContextGLGBM.cpp:
* Source/WebCore/platform/graphics/gbm/GraphicsContextGLGBM.h:
* Source/WebCore/platform/graphics/gbm/GraphicsContextGLGBMTextureMapper.cpp:
* Source/WebCore/platform/graphics/gbm/GraphicsContextGLGBMTextureMapper.h:
* Source/WebCore/platform/graphics/nicosia/texmap/NicosiaGCGLANGLELayer.cpp:
* Source/WebCore/platform/graphics/nicosia/texmap/NicosiaGCGLANGLELayer.h:
* Source/WebCore/platform/graphics/nicosia/texmap/NicosiaGCGLLayer.cpp: Removed.
* Source/WebCore/platform/graphics/nicosia/texmap/NicosiaGCGLLayer.h: Removed.
* Source/WebCore/platform/graphics/opengl/ExtensionsGLOpenGL.cpp: Removed.
* Source/WebCore/platform/graphics/opengl/ExtensionsGLOpenGL.h: Removed.
* Source/WebCore/platform/graphics/opengl/ExtensionsGLOpenGLCommon.cpp: Removed.
* Source/WebCore/platform/graphics/opengl/ExtensionsGLOpenGLCommon.h: Removed.
* Source/WebCore/platform/graphics/opengl/ExtensionsGLOpenGLES.cpp: Removed.
* Source/WebCore/platform/graphics/opengl/ExtensionsGLOpenGLES.h: Removed.
* Source/WebCore/platform/graphics/opengl/GraphicsContextGLOpenGL.cpp: Removed.
* Source/WebCore/platform/graphics/opengl/GraphicsContextGLOpenGL.h: Removed.
* Source/WebCore/platform/graphics/opengl/GraphicsContextGLOpenGLBase.cpp: Removed.
* Source/WebCore/platform/graphics/opengl/GraphicsContextGLOpenGLES.cpp: Removed.
* Source/WebCore/platform/graphics/texmap/BitmapTextureGL.h:
* Source/WebCore/platform/graphics/texmap/GraphicsContextGLTextureMapper.cpp: Removed.
* Source/WebCore/platform/graphics/texmap/GraphicsContextGLTextureMapper.h: Removed.
* Source/WebCore/platform/graphics/texmap/GraphicsContextGLTextureMapperANGLE.cpp:
* Source/WebCore/platform/graphics/texmap/GraphicsContextGLTextureMapperANGLE.h:
* Source/WebCore/platform/graphics/texmap/TextureMapperPlatformLayerBuffer.cpp:
(WebCore::TextureMapperPlatformLayerBuffer::paintToTextureMapper):
* Source/WebKit/GPUProcess/graphics/RemoteGraphicsContextGL.cpp:
* Source/cmake/OptionsGTK.cmake:
* Source/cmake/OptionsMac.cmake:
* Source/cmake/OptionsWPE.cmake:
* Source/cmake/OptionsWinCairo.cmake:

Canonical link: <a href="https://commits.webkit.org/254947@main">https://commits.webkit.org/254947@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ff38a6062efa63681c57852f98dc269357759374

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/90681 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/69/builds/35261 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/21279 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/99996 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/158318 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/94690 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/64/builds/33759 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/28894 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/83039 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/96395 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/96336 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/68/builds/26873 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/77504 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/26709 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/81649 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/81419 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/69736 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/1/builds/82222 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/34851 "Built successfully") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/15456 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/38/builds/77195 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/32663 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/16436 "Passed tests") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/46/builds/26639 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/3451 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/36429 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/39373 "Passed tests") | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/37/builds/79783 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/38350 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/35525 "Passed tests") | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/45/builds/17479 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

fujii commented on Source/cmake/OptionsGTK.cmake:
```diff
@@ -85,10 +84,6 @@ WEBKIT_OPTION_DEPEND(ENABLE_DOCUMENTATION ENABLE_INTROSPECTION)
 WEBKIT_OPTION_DEPEND(ENABLE_3D_TRANSFORMS USE_OPENGL_OR_ES)
 WEBKIT_OPTION_DEPEND(ENABLE_ASYNC_SCROLLING USE_OPENGL_OR_ES)
 WEBKIT_OPTION_DEPEND(ENABLE_GLES2 USE_OPENGL_OR_ES)
-WEBKIT_OPTION_DEPEND(ENABLE_WEBGL USE_OPENGL_OR_ES)
```
GTK port has `USE_OPENGL_OR_ES` build switch to disable OpenGL usage for some minor operating systems which has a problem for OpenGL.
I think they can't use WebGL even with ANGLE, `WEBKIT_OPTION_DEPEND(ENABLE_WEBGL USE_OPENGL_OR_ES)` shouldn't be removed.

mcatanzaro replied:
I understand this option has been broken for a while, though? Maybe it's time to just remove it?

But yeah, as long as the option still exists, the `WEBKIT_OPTION_DEPEND` should indeed not be removed. Good catch.

fujii replied:
I think someone fixed the problem very recently. https://github.com/WebKit/WebKit/pull/4035#discussion_r964183001

mcatanzaro replied:
OK, this line should remain then.

zdobersek replied:
It should remain. Same DEPEND should be added for WEBGL2, until it gets cleaned up (maybe by combining `ENABLE_WEBGL2` into `ENABLE_WEBGL`).

kkinnunen-apple replied:
thanks

fujii commented on Source/WebCore/html/canvas/ANGLEInstancedArrays.cpp:
```diff
@@ -49,12 +49,8 @@ WebGLExtension::ExtensionName ANGLEInstancedArrays::getName() const
-#if USE(ANGLE) || PLATFORM(GTK) || PLATFORM(WPE)
     return context.supportsExtension("GL_ANGLE_instanced_arrays"_s);
-#else
     UNUSED_PARAM(context);
```
UNUSED_PARAM(context); should be removed

zdobersek reviewed (approved):
LGTM

kkinnunen-apple commented on LayoutTests/platform/glib/TestExpectations:
```diff
@@ -1406,11 +1399,55 @@ webkit.org/b/229052 webgl/1.0.x/conformance/textures/misc/texture-corner-case-vi
+fast/canvas/webgl/webgl-compressed-texture-size-limit.html [ Failure ]
+fast/canvas/webgl/webgl-drawarrays-crash-2.html [ Failure ]
+fast/canvas/webgl/webgl-drawarrays-crash.html [ Failure ]
+fast/mediacapturefromelement/CanvasCaptureMediaStream-offscreencanvas.html [ Failure ]
```
since review, offscreen canvas tests added to failures here

kkinnunen-apple commented on Source/WebCore/html/canvas/WebGLRenderingContextBase.cpp:
```diff
@@ -924,10 +573,8 @@ std::unique_ptr<WebGLRenderingContextBase> WebGLRenderingContextBase::create(Can
-        // FIXME: OffscreenCanvas does not support GPU process.
-        context = createWebProcessGraphicsContextGL(attributes);
-    }
+    // FIXME: OffscreenCanvas does not support GPU process and ANGLE does not support 
```
@zdobersek since review, I had to disable offscreen canvas. It won't work in-process with ANGLE and we haven't yet fixed needed stuff for it to work with GPUP. 
So to get offscreen canvas back, we'll work implementing it in GPUP. Still ok?

zdobersek replied:
OK.

kkinnunen-apple committed:
WebGL has a lot of #if USE(ANGLE) blocks, making it hard to maintain
https://bugs.webkit.org/show_bug.cgi?id=229405
rdar://problem/82517622

Reviewed by Žan Doberšek.

Remove non-ANGLE WebGL implementations. This makes ANGLE the only WebGL backend
available. WebGL specific alterations to OpenGL semantics is implemented
correctly only in ANGLE. The OpenGL and OpenGL ES backends were not
being updated to match the WebGL specification or passing the tests.

Remove USE_ANGLE from WebKit level. Use USE_ANGLE for the ports
that want to control whether OpenGL is ANGLE or OpenGL is the platform
OpenGL.

ENABLE(WEBGL) && USE(ANGLE) is now ENABLE(WEBGL).

* Source/ThirdParty/ANGLE/CMakeLists.txt:
* Source/ThirdParty/ANGLE/PlatformGTK.cmake:
* Source/ThirdParty/ANGLE/PlatformWPE.cmake:
* Source/ThirdParty/ANGLE/include/CMakeLists.txt:
* Source/WTF/wtf/Platform.h:
* Source/WebCore/CMakeLists.txt:
* Source/WebCore/Modules/webxr/WebXROpaqueFramebuffer.cpp:
(WebCore::WebXROpaqueFramebuffer::startFrame):
(WebCore::WebXROpaqueFramebuffer::endFrame):
(WebCore::WebXROpaqueFramebuffer::setupFramebuffer):
* Source/WebCore/Modules/webxr/WebXRWebGLLayer.cpp:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/html/canvas/ANGLEInstancedArrays.cpp:
(WebCore::ANGLEInstancedArrays::supported):
* Source/WebCore/html/canvas/WebGLCompressedTextureS3TC.cpp:
(WebCore::WebGLCompressedTextureS3TC::supported):
* Source/WebCore/html/canvas/WebGLDrawBuffers.cpp:
(WebCore::WebGLDrawBuffers::supported):
* Source/WebCore/html/canvas/WebGLFramebuffer.cpp:
(WebCore::WebGLFramebuffer::setAttachmentForBoundFramebuffer):
(WebCore::WebGLFramebuffer::removeAttachmentFromBoundFramebuffer):
(WebCore::getClearBitsByAttachmentType): Deleted.
(WebCore::getClearBitsByFormat): Deleted.
(WebCore::isAttachmentComplete): Deleted.
(WebCore::WebGLFramebuffer::getColorBufferWidth const): Deleted.
(WebCore::WebGLFramebuffer::getColorBufferHeight const): Deleted.
(WebCore::WebGLFramebuffer::getColorBufferFormat const): Deleted.
(WebCore::WebGLFramebuffer::checkStatus const): Deleted.
(WebCore::WebGLFramebuffer::onAccess): Deleted.
(WebCore::WebGLFramebuffer::initializeAttachments): Deleted.
* Source/WebCore/html/canvas/WebGLFramebuffer.h:
* Source/WebCore/html/canvas/WebGLProgram.cpp:
(WebCore::WebGLProgram::cacheActiveAttribLocations):
* Source/WebCore/html/canvas/WebGLRenderingContext.cpp:
(WebCore::WebGLRenderingContext::initializeVertexArrayObjects):
* Source/WebCore/html/canvas/WebGLRenderingContextBase.cpp:
(WebCore::WebGLRenderingContextBase::initializeNewContext):
(WebCore::WebGLRenderingContextBase::setupFlags):
(WebCore::WebGLRenderingContextBase::~WebGLRenderingContextBase):
(WebCore::WebGLRenderingContextBase::reshape):
(WebCore::WebGLRenderingContextBase::bindTexture):
(WebCore::WebGLRenderingContextBase::checkFramebufferStatus):
(WebCore::WebGLRenderingContextBase::clear):
(WebCore::WebGLRenderingContextBase::compileShader):
(WebCore::WebGLRenderingContextBase::compressedTexImage2D):
(WebCore::WebGLRenderingContextBase::compressedTexSubImage2D):
(WebCore::WebGLRenderingContextBase::copyTexSubImage2D):
(WebCore::WebGLRenderingContextBase::deleteTexture):
(WebCore::WebGLRenderingContextBase::disable):
(WebCore::WebGLRenderingContextBase::disableVertexAttribArray):
(WebCore::WebGLRenderingContextBase::validateVertexAttributes):
(WebCore::WebGLRenderingContextBase::drawArrays):
(WebCore::WebGLRenderingContextBase::drawElements):
(WebCore::WebGLRenderingContextBase::enable):
(WebCore::WebGLRenderingContextBase::enableVertexAttribArray):
(WebCore::WebGLRenderingContextBase::finish):
(WebCore::WebGLRenderingContextBase::flush):
(WebCore::WebGLRenderingContextBase::framebufferRenderbuffer):
(WebCore::WebGLRenderingContextBase::framebufferTexture2D):
(WebCore::WebGLRenderingContextBase::generateMipmap):
(WebCore::WebGLRenderingContextBase::getProgramParameter):
(WebCore::WebGLRenderingContextBase::getUniform):
(WebCore::WebGLRenderingContextBase::getUniformLocation):
(WebCore::WebGLRenderingContextBase::getVertexAttrib):
(WebCore::WebGLRenderingContextBase::linkProgramWithoutInvalidatingAttribLocations):
(WebCore::WebGLRenderingContextBase::readPixels):
(WebCore::WebGLRenderingContextBase::renderbufferStorage):
(WebCore::WebGLRenderingContextBase::shaderSource):
(WebCore::WebGLRenderingContextBase::texImageSourceHelper):
(WebCore::WebGLRenderingContextBase::texImageArrayBufferViewHelper):
(WebCore::WebGLRenderingContextBase::texImageImpl):
(WebCore::WebGLRenderingContextBase::texImage2DBase):
(WebCore::WebGLRenderingContextBase::texSubImage2DBase):
(WebCore::WebGLRenderingContextBase::validateTexFunc):
(WebCore::WebGLRenderingContextBase::validateTexFuncParameters):
(WebCore::WebGLRenderingContextBase::validateTexFuncFormatAndType):
(WebCore::WebGLRenderingContextBase::validateForbiddenInternalFormats):
(WebCore::WebGLRenderingContextBase::copyTexImage2D):
(WebCore::WebGLRenderingContextBase::texParameter):
(WebCore::WebGLRenderingContextBase::uniform1i):
(WebCore::WebGLRenderingContextBase::uniform1iv):
(WebCore::WebGLRenderingContextBase::validateTextureBinding):
(WebCore::WebGLRenderingContextBase::validateTexture2DBinding):
(WebCore::WebGLRenderingContextBase::vertexAttribfImpl):
(WebCore::WebGLRenderingContextBase::vertexAttribfvImpl):
(WebCore::WebGLRenderingContextBase::setFramebuffer):
(WebCore::WebGLRenderingContextBase::restoreCurrentTexture2D):
(WebCore::WebGLRenderingContextBase::drawArraysInstanced):
(WebCore::WebGLRenderingContextBase::drawElementsInstanced):
(WebCore::clip2D): Deleted.
(WebCore::getChannelBitsByFormat): Deleted.
(WebCore::possibleFormatAndTypeForInternalFormat): Deleted.
(WebCore::WebGLRenderingContextBase::validateNPOTTextureLevel): Deleted.
(WebCore::WebGLRenderingContextBase::validateDrawArrays): Deleted.
(WebCore::WebGLRenderingContextBase::validateDrawElements): Deleted.
(WebCore::internalFormatTheme): Deleted.
(WebCore::numberOfComponentsForFormat): Deleted.
(WebCore::numberOfComponentsForInternalFormat): Deleted.
(WebCore::isRGBFormat): Deleted.
(WebCore::WebGLRenderingContextBase::checkTextureCompleteness): Deleted.
(WebCore::WebGLRenderingContextBase::createFallbackBlackTextures1x1): Deleted.
(WebCore::WebGLRenderingContextBase::isTexInternalFormatColorBufferCombinationValid): Deleted.
(WebCore::WebGLRenderingContextBase::getBoundReadFramebufferColorFormat): Deleted.
(WebCore::WebGLRenderingContextBase::getBoundReadFramebufferWidth): Deleted.
(WebCore::WebGLRenderingContextBase::getBoundReadFramebufferHeight): Deleted.
(WebCore::calculateBytesForASTC): Deleted.
(WebCore::WebGLRenderingContextBase::validateCompressedTexFuncData): Deleted.
(WebCore::WebGLRenderingContextBase::validateCompressedTexDimensions): Deleted.
(WebCore::WebGLRenderingContextBase::validateCompressedTexSubDimensions): Deleted.
(WebCore::WebGLRenderingContextBase::initVertexAttrib0): Deleted.
(WebCore::WebGLRenderingContextBase::validateSimulatedVertexAttrib0): Deleted.
(WebCore::WebGLRenderingContextBase::simulateVertexAttrib0): Deleted.
(WebCore::WebGLRenderingContextBase::restoreStatesAfterVertexAttrib0Simulation): Deleted.
(WebCore::WebGLRenderingContextBase::applyStencilTest): Deleted.
* Source/WebCore/html/canvas/WebGLRenderingContextBase.h:
* Source/WebCore/html/canvas/WebGLShader.h:
* Source/WebCore/html/canvas/WebGLTexture.cpp:
(WebCore::WebGLTexture::WebGLTexture):
(WebCore::WebGLTexture::setTarget):
(WebCore::WebGLTexture::setParameteri): Deleted.
(WebCore::WebGLTexture::setParameterf): Deleted.
(WebCore::WebGLTexture::setLevelInfo): Deleted.
(WebCore::WebGLTexture::generateMipmapLevelInfo): Deleted.
(WebCore::WebGLTexture::getInternalFormat const): Deleted.
(WebCore::WebGLTexture::getType const): Deleted.
(WebCore::WebGLTexture::getWidth const): Deleted.
(WebCore::WebGLTexture::getHeight const): Deleted.
(WebCore::WebGLTexture::isValid const): Deleted.
(WebCore::WebGLTexture::markInvalid): Deleted.
(WebCore::WebGLTexture::isNPOT): Deleted.
(WebCore::WebGLTexture::isNPOT const): Deleted.
(WebCore::WebGLTexture::needToUseBlackTexture const): Deleted.
(WebCore::WebGLTexture::canGenerateMipmaps): Deleted.
(WebCore::WebGLTexture::isCompressed const): Deleted.
(WebCore::WebGLTexture::setCompressed): Deleted.
(WebCore::WebGLTexture::mapTargetToIndex const): Deleted.
(WebCore::internalFormatIsFloatType): Deleted.
(WebCore::internalFormatIsHalfFloatType): Deleted.
(WebCore::WebGLTexture::update): Deleted.
(WebCore::WebGLTexture::getLevelInfo const): Deleted.
* Source/WebCore/html/canvas/WebGLTexture.h:
* Source/WebCore/html/canvas/WebGLVertexArrayObjectBase.cpp:
(WebCore::WebGLVertexArrayObjectBase::WebGLVertexArrayObjectBase):
(WebCore::WebGLVertexArrayObjectBase::unbindBuffer):
(WebCore::WebGLVertexArrayObjectBase::updateVertexAttrib0): Deleted.
* Source/WebCore/html/canvas/WebGLVertexArrayObjectBase.h:
* Source/WebCore/platform/TextureMapper.cmake:
* Source/WebCore/platform/graphics/ANGLEWebKitBridge.cpp: Removed.
* Source/WebCore/platform/graphics/ANGLEWebKitBridge.h: Removed.
* Source/WebCore/platform/graphics/GraphicsContextGL.cpp:
* Source/WebCore/platform/graphics/angle/ANGLEHeaders.h:
* Source/WebCore/platform/graphics/angle/ANGLEUtilities.cpp:
* Source/WebCore/platform/graphics/angle/ANGLEUtilities.h:
* Source/WebCore/platform/graphics/angle/GraphicsContextGLANGLE.cpp:
* Source/WebCore/platform/graphics/angle/GraphicsContextGLANGLE.h:
* Source/WebCore/platform/graphics/cg/GraphicsContextGLCG.cpp:
* Source/WebCore/platform/graphics/cocoa/ANGLEUtilitiesCocoa.cpp:
* Source/WebCore/platform/graphics/cocoa/ANGLEUtilitiesCocoa.h:
* Source/WebCore/platform/graphics/gbm/GraphicsContextGLANGLELinux.cpp:
* Source/WebCore/platform/graphics/gbm/GraphicsContextGLFallback.cpp:
* Source/WebCore/platform/graphics/gbm/GraphicsContextGLFallback.h:
* Source/WebCore/platform/graphics/gbm/GraphicsContextGLGBM.cpp:
* Source/WebCore/platform/graphics/gbm/GraphicsContextGLGBM.h:
* Source/WebCore/platform/graphics/gbm/GraphicsContextGLGBMTextureMapper.cpp:
* Source/WebCore/platform/graphics/gbm/GraphicsContextGLGBMTextureMapper.h:
* Source/WebCore/platform/graphics/nicosia/texmap/NicosiaGCGLANGLELayer.cpp:
* Source/WebCore/platform/graphics/nicosia/texmap/NicosiaGCGLANGLELayer.h:
* Source/WebCore/platform/graphics/nicosia/texmap/NicosiaGCGLLayer.cpp: Removed.
* Source/WebCore/platform/graphics/nicosia/texmap/NicosiaGCGLLayer.h: Removed.
* Source/WebCore/platform/graphics/opengl/ExtensionsGLOpenGL.cpp: Removed.
* Source/WebCore/platform/graphics/opengl/ExtensionsGLOpenGL.h: Removed.
* Source/WebCore/platform/graphics/opengl/ExtensionsGLOpenGLCommon.cpp: Removed.
* Source/WebCore/platform/graphics/opengl/ExtensionsGLOpenGLCommon.h: Removed.
* Source/WebCore/platform/graphics/opengl/ExtensionsGLOpenGLES.cpp: Removed.
* Source/WebCore/platform/graphics/opengl/ExtensionsGLOpenGLES.h: Removed.
* Source/WebCore/platform/graphics/opengl/GraphicsContextGLOpenGL.cpp: Removed.
* Source/WebCore/platform/graphics/opengl/GraphicsContextGLOpenGL.h: Removed.
* Source/WebCore/platform/graphics/opengl/GraphicsContextGLOpenGLBase.cpp: Removed.
* Source/WebCore/platform/graphics/opengl/GraphicsContextGLOpenGLES.cpp: Removed.
* Source/WebCore/platform/graphics/texmap/BitmapTextureGL.h:
* Source/WebCore/platform/graphics/texmap/GraphicsContextGLTextureMapper.cpp: Removed.
* Source/WebCore/platform/graphics/texmap/GraphicsContextGLTextureMapper.h: Removed.
* Source/WebCore/platform/graphics/texmap/GraphicsContextGLTextureMapperANGLE.cpp:
* Source/WebCore/platform/graphics/texmap/GraphicsContextGLTextureMapperANGLE.h:
* Source/WebCore/platform/graphics/texmap/TextureMapperPlatformLayerBuffer.cpp:
(WebCore::TextureMapperPlatformLayerBuffer::paintToTextureMapper):
* Source/WebKit/GPUProcess/graphics/RemoteGraphicsContextGL.cpp:
* Source/cmake/OptionsGTK.cmake:
* Source/cmake/OptionsMac.cmake:
* Source/cmake/OptionsWPE.cmake:
* Source/cmake/OptionsWinCairo.cmake:

Canonical link: https://commits.webkit.org/254947@main
